pydantic/types.py

```
@@ -626,3 +634,238 @@ def display(self) -> str:
+            if self._color_match is not None:
+                return
+
+            # hex string
+            self._color_match = self._parse_hex_str(color)
+            if self._color_match is not None:
+                return
+
+            # named colour
+            if color in colors.BY_NAME:
+                self._color_match = color
+                return
+
+    def _get_color_or_raise(self) -> None:
+        """
+        Raise error if color name not found
+        """
+        if self._color_match is None:
+            raise errors.ColorError()
+
+    def original(self) -> ColorType:
+        """
+        Return original value passed to the model
+        """
+        return self._original
+
+    def as_hex(self, prefix: str = '0x', reduce: bool = True) -> str:
+        """
+        Return hexadecimal value of the color
+
+        Try reduce hex code to 3-digit format, fallback to 6-digit.
+        """
+        self._get_color_or_raise()
+        hexadecimal, _ = colors.BY_NAME[self._color_match or '']
+        return '{}{}'.format(prefix, self.reduce_6_digit_hex(hexadecimal) if reduce else hexadecimal)
+
+    def as_tuple(self, alpha: bool = False) -> AnyRGBType:
+        """
+        Return RGB or RGBA tuple
+        """
+        self._get_color_or_raise()
+        _, rgb = colors.BY_NAME[self._color_match or '']
+        r, g, b = rgb
+
+        if alpha:
+            return r, g, b, 1.0
+        return r, g, b
+
+    def as_rgb(self, alpha: bool = False) -> str:
+        """
+        Return RGB or RGBA string representation of the color
+        """
+        rgb = self.as_tuple(alpha)
+        if alpha:
+            return 'rgba({}, {}, {}, {})'.format(*rgb)
+        return 'rgb({}, {}, {})'.format(*rgb)
+
+    def as_named_color(self) -> str:
+        """
+        Return name of the color as per CSS3 specification.
+        """
+        self._get_color_or_raise()
+        return self._color_match  # type: ignore
```

I think this is wrong, it should raise an error if there isn't a named colour, eg. `#bad` is a valid colour, but not a named colour (I assume).

I these the error that should be raised should be `ColorError` but with a custom message "color is valid but does not have a name" or whatever.

---

I have implemented check for a valid but unnamed colour. Hope I got it right